File: Java/civitas/src/civitas/SorpresaEspeculador.java
package civitas;

import java.util.ArrayList;

public class SorpresaEspeculador extends Sorpresa{
    private int valor;
    
    SorpresaEspeculador(int _valor){
        super("Soy Especulador");
        valor=_valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(actual, todos)){
            super.informe(actual, todos);
            Especulador especulador = new Especulador(todos.get(actual), valor);
            todos.set(actual, especulador);
        }
    }
}

File: Ruby/civitas/lib/sorpresa_especulador.rb
#encoding: utf­8
require_relative "sorpresa.rb"
class SorpresaEspeculador < Sorpresa
  def initialize(_valor)
    super("Soy un especulador")
    @valor = _valor
  end
  
  def aplicar_a_jugador(actual, todos)
    if(super.jugador_correcto(actual, todos))
      super.informe(actual, todos)
      especulador = Especulador.new(todos.at(actual), @valor)
      #todos.set()
    end
  end
  
  def to_ss()
    return @texto
  end
end
